### PR TITLE
Check for unused and undefined parameters

### LINKED ISSUE #84: Check unused and undefined parameters
I had a typo so that parameter name differed between definition and usage sites. It would be nice if the library noticed that.

### PR DESCRIPTION
**Pull request by ethercrow**
Fixes #84

### CODE REVIEW

**Comment by ethercrow** [comment]
I'm not sure about using `println` to indicate warnings, but since nobody looks at the value returned by `validate`, is there a better alternative?

File: src/io/sarnowski/swagger1st/validation.clj [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  (let [used-params (collect-used-params definition)
+        defined-params (collect-defined-params definition)
+        used-but-undefined-params (clojure.set/difference used-params defined-params)
+        defined-but-unused-params (clojure.set/difference defined-params used-params)
+        ]
+    (if-not (empty? used-but-undefined-params)
+      (throw (ex-info "Some parameters are used but not defined" {:names used-but-undefined-params})))
+    (if-not (empty? defined-but-unused-params)
+      ;; Not a critical error, a warning is enough
+      (println "[swagger1st] [WARN] Params defined but not used: " defined-but-unused-params))))
```
Comment (dryewo) [inline src/io/sarnowski/swagger1st/validation.clj]: please log a warning

File: src/io/sarnowski/swagger1st/validation.clj [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+      (map #(clojure.string/replace % "#/parameters/" ""))
+      set)))
+
+(defn validate [definition]
+  (schema/validate swagger-2-0/root-object definition)
+  (let [used-params (collect-used-params definition)
+        defined-params (collect-defined-params definition)
+        used-but-undefined-params (clojure.set/difference used-params defined-params)
+        defined-but-unused-params (clojure.set/difference defined-params used-params)
+        ]
```
Comment (dryewo) [inline src/io/sarnowski/swagger1st/validation.clj]: hanging `]`

File: src/io/sarnowski/swagger1st/validation.clj [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    (if-not (empty? used-but-undefined-params)
```
Comment (dryewo) [inline src/io/sarnowski/swagger1st/validation.clj]: I suggest `when` instead of `if` with one branch

**Comment by ethercrow** [comment]
@dryewo updated

**Comment by dryewo** [comment]
:+1:

**Comment by ethercrow** [comment]
👍

**Pull request merged**